Source/CesiumRuntime/Private/CesiumTextureUtility.cpp

```diff
@@ -1,18 +1,351 @@
+    }
+
+    const uint32 SrcPitch =
+        NumColumns * BlockBytes; // Num-of bytes per row in the source data
+    const uint32 EffectiveSize = BlockBytes * NumColumns * NumRows;
+
+    // Copy the texture data.
+    CopyTextureData2D(pSrcData, Dest, mipHeight, format, SrcPitch, DestPitch);
+  }
+}
+
+/**
+ * @brief A wrapper for CesiumGltf::ImageCesium that allows it to be used as
+ * bulk data for RHI texture creation. This prevents having to copy texture
+ * data to an intermediary, Unreal-specific CPU buffer before being copied to
+ * the GPU (in particular FTexture2DMipMap).
+ */
+class FCesiumTextureData : public FResourceBulkDataInterface {
+public:
+  FCesiumTextureData(const CesiumGltf::ImageCesium& image)
+      : _textureData(image.pixelData.data()),
+        _textureDataSize(
+            (!image.mipPositions.empty())
+                ? static_cast<uint32>(image.mipPositions[0].byteSize)
+                : static_cast<uint32>(image.pixelData.size())) {}
+
+  const void* GetResourceBulkData() const override {
+    return (void*)_textureData;
+  }
+
+  uint32 GetResourceBulkDataSize() const override { return _textureDataSize; }
+
+  void Discard() override {}
+
+private:
+  const std::byte* _textureData;
+  uint32 _textureDataSize;
+};
+
+/**
+ * @brief An RHI resource that creates and destroys RHI textures. If a non-null
+ * FTexture2DRHIRef is given, ownership of it is assumed and it will be
+ * destroyed in ReleaseResource. Otherwise, the RHI texture will be created
+ * from the in-memory Cesium glTF image in InitRHI (on the render thread).
+ *
+ */
+class FCesiumTextureResource : public FTextureResource {
+public:
+  FCesiumTextureResource(
+      UTexture* pTexture,
+      const CesiumGltf::ImageCesium* pCesiumImage,
+      FTexture2DRHIRef rhiTextureRef,
+      EPixelFormat format,
+      bool sRGB,
+      bool generateMipMaps,
+      uint32 extData)
+      : _pTexture(pTexture),
+        // TODO: ImageCesium is copied for debugging purposes,
+        // why is keeping around a pointer not working?
+        _image(*pCesiumImage),
+        _pCesiumImage(&_image),
+        _format(format),
+        _generateMipMaps(generateMipMaps),
+        _platformExtData(extData) {
+    this->bGreyScaleFormat = (_format == PF_G8) || (_format == PF_BC4);
+    this->bSRGB = sRGB;
+    // Will be null if async texture creation was unavailable.
+    this->TextureRHI = rhiTextureRef;
+  }
+
+  virtual ~FCesiumTextureResource() {
+    check(_pTexture != nullptr);
+    _pTexture->SetResource(nullptr);
+  }
+
+  uint32 GetSizeX() const override {
+    return static_cast<uint32>(_pCesiumImage->width);
+  }
+
+  uint32 GetSizeY() const override {
+    return static_cast<uint32>(_pCesiumImage->height);
+  }
+
+  virtual void InitRHI() override {
+    // TODO: does anisotropy work?
+    FSamplerStateInitializerRHI samplerStateInitializer(
+        SF_Trilinear,
+        AM_Wrap,
+        AM_Wrap,
+        AM_Wrap,
```

Comment:
Using wrap here instead of clamp is likely the cause of the seams I'm seeing between raster overlay tiles.
![image](https://user-images.githubusercontent.com/924374/193195592-c42052fe-faa6-496a-bf77-d6d3408ad75d.png)

Reply:
But wrap is the right answer in other scenarios, of course. It needs to be passed in from the sampler.

Reply:
This should also be fixed now!